### DISCUSSION TITLE
Translating client app using LocaleAPI

### DISCUSSION THREAD

**Discussion by Meldiron** [appwrite/appwrite#1401] [Ideas]
Since we have a LocaleAPI that can provide useful information about countries, phone numbers and language codes could finish the full circle and get an API to locale our own texts? This shouldn't be hard to implement, it could simply be an internal collection.

In SDK, we could have something like `this.sdk.locale.getTranslation("my-text-key");` that would get a message with this customID translated into the language I currently have selected (or fallback to default language).

In administration, there could be new "Translations" section where we can define a key-value pair where the key is customID and value is the translation itself. This could have multiple tabs for multiple languages we translate our application to.

Finally, we need some way to use placeholder, pluralization and date localization, but I believe this is a part of some protocol (i18n?) so it should not be a problem to research.

In future, we could get implementations into specific frameworks, such as React or Angular. In angular, we could do this in HTML code: `<p> {{ 'footer-fulltext' | appwrite }} </p>`.

---

What does community think about this?

**Comment by Maatteogekko** [comment]
I would really love to see this in Appwrite. The more features the better! 

For example, in the case of a mobile app, the first time it is opened I would check the locale and download all the localised texts from appwrite. In this way I would be able to outsource the translation logic outside the app, and get the app size down a little along the way, since I would not need to bundle the translations with the app.

This is just the first use case that came to my mind, I am sure there are many more
